src/lxc/conf.c

```diff
@@ -3323,28 +3323,95 @@ int lxc_map_ids(struct lxc_list *idmap, pid_t pid)
 {
 	struct id_map *map;
 	struct lxc_list *iterator;
+	struct stat sb;
 	enum idtype type;
 	char *pos;
-	char *buf = NULL, *cmdpath = NULL;
+	int euid;
 	bool use_shadow = false;
-	int ret = 0;
+	int fret = 0, ret = 0;
+	char *buf = NULL, *newuidmap_path, *newgidmap_path;
 
-	/*
-	 * If newuidmap exists, that is, if shadow is handing out subuid
-	 * ranges, then insist that root also reserve ranges in subuid.  This
+	euid = geteuid();
+
+	/* If new{g,u}idmap exists, that is, if shadow is handing out subuid
+	 * ranges, then insist that root also reserve ranges in subuid. This
 	 * will protected it by preventing another user from being handed the
 	 * range by shadow.
 	 */
-	cmdpath = on_path("newuidmap", NULL);
-	if (cmdpath) {
-		use_shadow = true;
-		free(cmdpath);
+	newuidmap_path = on_path("newuidmap", NULL);
+	if (newuidmap_path) {
+		bool has_sufficient_privilege = false;
```

Comment:
So if you turn that block into a helper then it goes away? :)

In either case, the check here to break out of the loop needs to be for fret right?  or not?